### ISSUE TITLE
Merge TriggerMetadata and AuthParams into one concept

### ISSUE THREAD

**Issue by coderanger** [kedacore/keda#1793]
### Proposal

Right now we have two `map[string]string` for each trigger, one is the "metadata", these are coded into the ScaledObject/Job, the other is authentication parameters from The TriggerAuthentication and ClusterTriggerAuthentication types. While some scalers have code to allow some settings to be pulled from either, this is inconsistent. And in general having two parallel data sets increases cognitive complexity as both as a user and a developer. I think we can merge these together into one system, with values set from both the trigger spec and authentication data compiled into one map that gets handed to the scaler code. This would allow increased flexibility to users who need complex or fancy setups (want to pull your whole trigger config from a custom Vault plugin? go for it!), reduce confusion with scalers that allow certain settings in only one place or the other, and generally streamline our codebase.

### Use-Case

Mostly covered above but some concrete examples:

1. Local dev where I just want to put the password or token directly into a trigger spec while I'm testing my code.
2. Weird fancy setups that need more dynamic trigger configuration than currently expected.
3. Scaler authors that don't know to make things readable from both maps and so products scalers that break expectations.

**Comment by tomkerkhove** [comment]
I'm not sure if I fully get your proposal in the sense that it's just internally merging things in the code, a separate way to configure or something else.

I think it's purely internally but just verifying.

**Comment by coderanger** [comment]
Yes, purely internal. I think in the future we could allow for something like embedding a triggerauth spec directly into a trigger definition but that's separate. This would be changing the API exposed to scalers, basically just taking the two existing maps and merging them into one `TriggerSettings map[string]string` or something, and then shifting all scalers to read from that instead of the two existing maps.

**Comment by zroubalik** [comment]
+1 for internal change towards unified map/config/whatnot. My only concern is actually that we would allow specifying credentials in the plaintext and not in the Secret or Vault or whatnot. Which is imho not a good practise.

**Comment by coderanger** [comment]
Unstale, this really should happen some day :)
